modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/schema/SqlSchemaManagerImpl.java

```diff
@@ -64,6 +76,30 @@ public SchemaPlus schema(@Nullable String schema) {
         return schema != null ? calciteSchema.getSubSchema(schema) : calciteSchema;
     }
 
+    /** {@inheritDoc} */
+    @Override
+    @NotNull
+    public InternalIgniteTable tableById(IgniteUuid id) {
+        ensureTableUpToDate(id);
+
+        InternalIgniteTable table = tablesById.get(id);
+
+        if (table == null) {
+            throw new IgniteInternalException(
+                    LoggerMessageHelper.format("Table not found [tableId={}]", id));
+        }
+
+        return table;
+    }
+
+    private void ensureTableUpToDate(IgniteUuid id) {
```

Comment:
"UpToDate" confusing a bit, everything is relative ) possibly : ensureTableOperable ?

Reply:
What do you think about `ensureTableStructuresCteated`?

Reply:
fixed